mlflow/tensorflow/__init__.py

```diff
@@ -818,9 +866,45 @@ def log_model_without_starting_new_run():
                         tf_meta_graph_tags=[tag_constants.SERVING],
                         tf_signature_def_key="predict",
                     )
-                    save_model(path=local_path, mlflow_model=mlflow_model, **save_model_kwargs)
-                    client = MlflowClient()
-                    client.log_artifacts(_AUTOLOG_RUN_ID, local_path, artifact_path)
+
+                    input_example = None
+                    signature = None
+                    if log_input_examples:
+
+                        def predict_input_fn():
+                            """
+                            Builds an input function to be used for tf's predict
+                            in order to get predicted values required
+                            for the model signature inference.
+                            """
+                            input_slices = input_example_slice
+                            if isinstance(input_example_slice, dict):
+                                input_slices = {
+                                    k: tensorflow.convert_to_tensor(v)
+                                    for k, v in input_example_slice.items()
+                                }
+                            return tensorflow.data.Dataset.from_tensor_slices(input_slices).batch(1)
+
+                        predicted_values = list(tf_model.predict(predict_input_fn))
+
+                        input_example, signature = resolve_input_example_and_signature(
+                            lambda: input_example_slice,
+                            lambda in_ex: infer_signature(input_example_slice, predicted_values[0]),
```

Comment:
If `input_example_slice` is `None`, I think this will print a strange error - e.g. "cannot concatenate `NoneType` and `string`." This can be `None` if `_extract_data_from_input_fn` doesn't know how to handle the data type.

Instead of passing `lambda: input_example_slice`, can we pass a function that either returns `input_example_slice` if it is `not None` or throws?

E.g.

```
def get_input_example_slice():
    if input_example_slice is None:
        raise MlflowException(
            "TensorFlow autologging can only log input examples for inputs of the following types: ..."
        )
    return input_example_slice

```

Reply:
@bali0019 Can we address this comment?